src/test/java/org/jenkinsci/test/acceptance/recorder/TestRecorderRuleTest.java

```diff
@@ -0,0 +1,106 @@
+package org.jenkinsci.test.acceptance.recorder;
+
+import org.hamcrest.core.Is;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+import org.junit.rules.TestWatcher;
+import org.junit.runner.Description;
+
+import java.io.File;
+
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+
+public class TestRecorderRuleTest {
+
+    @Test
+    public void shouldNotRecordSuccessTestExecutionByDefault() {
+        TestRecorderRule testRecorderRule = new TestRecorderRule();
+        Description shouldNotRecordSuccessTestExecutionByDefault = Description.createTestDescription("org.jenkinsci.test.acceptance.recorder.TestRecorderRuleTest", "shouldNotRecordSuccessTestExecutionByDefault");
+        testRecorderRule.starting(shouldNotRecordSuccessTestExecutionByDefault);
+
+        System.out.println("Hello World");
+
+        testRecorderRule.succeeded(shouldNotRecordSuccessTestExecutionByDefault);
+        testRecorderRule.finished(shouldNotRecordSuccessTestExecutionByDefault);
+
+        File outputFile = new File("target", "org.jenkinsci.test.acceptance.recorder.TestRecorderRuleTest-shouldNotRecordSuccessTestExecutionByDefault.mov");
+        assertThat(outputFile.exists(), is(false));
+
+        //Clean the field
+        outputFile.delete();
+    }
+    @Test
+    public void shouldRecordFailingTestExecutionByDefault() {
+
+        TestRecorderRule testRecorderRule = new TestRecorderRule();
+        Description shouldNotRecordSuccessTestExecutionByDefault = Description.createTestDescription("org.jenkinsci.test.acceptance.recorder.TestRecorderRuleTest", "shouldRecordFailingTestExecutionByDefault");
+        testRecorderRule.starting(shouldNotRecordSuccessTestExecutionByDefault);
+
+        System.out.println("Good Bye World");
+        //succeeded is not called since a failure is simulated
+        //testRecorderRule.succeeded(shouldNotRecordSuccessTestExecutionByDefault);
+        testRecorderRule.finished(shouldNotRecordSuccessTestExecutionByDefault);
+
+        File outputFile = new File("target", "org.jenkinsci.test.acceptance.recorder.TestRecorderRuleTest-shouldRecordFailingTestExecutionByDefault.mov");
+        assertThat(outputFile.exists(), is(true));
+
+        //Clean the field
+        outputFile.delete();
+    }
+
+    @Test
+    public void shouldRecordSuccessTestExecutionWhenSaveAll() {
+        String oldValue = System.getProperty("RECORDER_SAVE_ALL");
+        System.setProperty("RECORDER_SAVE_ALL", "true");
```

Comment:
Perhaps I miss something, but when I tried this it recorded my current content of the screen and not the firefox window. If this happens when running in parallel (and that is what we rely on during CI build), I guess the test runs will fight over the screen making at least `N-1/N` of the recording useless or even confusing as it will record some other test.

I wanted to suggest to make this feature turned off by default. Perhaps it should run only when `forkCount == 1`.